pkg/liqoctl/add/add_test.go

```diff
@@ -0,0 +1,100 @@
+package add
+
+import (
+	"context"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/types"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/utils/pointer"
+
+	discoveryv1alpha1 "github.com/liqotech/liqo/apis/discovery/v1alpha1"
+	"github.com/liqotech/liqo/pkg/consts"
+	"github.com/liqotech/liqo/pkg/utils/authenticationtoken"
+	foreigncluster "github.com/liqotech/liqo/pkg/utils/foreignCluster"
+)
+
+func TestAddCommand(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Test Parameters Fetching")
+}
+
+var _ = Describe("Extract elements from apiServer", func() {
+	DescribeTable("Ensure correct resources are created or an exception is raised",
+		func(t *ClusterArgs, result types.GomegaMatcher, message string) {
+			// Set up the cluster install
+			Expect(setUpEnvironment(ctx, clientset, t)).ToNot(HaveOccurred())
```

Comment:
```suggestion
			Expect(setUpEnvironment(ctx, clientset, t)).To(Succeed())
```

Would move this to a BeforeEach block, with a corresponding AfterEach one to clean the environment, to avoid interference between one entry and the other.